docs/prompting/self_criticism/reversecot.md

```diff
@@ -1,7 +1,263 @@
+):
+    formatted_original_conditions = "\n- ".join(original_condition_list)
+    formatted_final_conditions = "\n- ".join(final_condition_list)
+    return client.chat.completions.create(
+        model="gpt-4o",
+        response_model=ModelFeedback,
+        messages=[
+            {
+                "role": "system",
+                "content": f"""
+                You are an expert AI system that excels at
+                analyzing and comparing two lists of conditions.
+
+                Original Condition List:
+                {formatted_original_conditions}
+
+                Reconstructed Condition List:
+                {formatted_final_conditions}
+
+                Determine if the two condition lists are roughly
+                equivalent. If they are not, give targetted
+                feedback on what is missing from the reconstructed
+                condition list as compared to the original condition
+                list and how it can be fixed.
+                """,
+            }
+        ],
+    )
+
+
+def revise_response(response: ModelResponse, feedback: ModelFeedback):
+    formatted_inconsistencies = "\n- ".join(feedback.detected_inconsistencies)
+    return client.chat.completions.create(
+        model="gpt-4o",
+        messages=[
+            {
+                "role": "system",
+                "content": f"""
+                Here are the mistakes and reasons in your answer
+                to the prompt
+
+                Original Response: {response.correct_answer}
+                You have overlooked some real conditions:
+                {formatted_inconsistencies}
+
+                Here are detailed reasons:
+                {feedback.feedback}
+
+                Generate a revised response that takes into account
+                the detailed feedback and includes the ignored
+                conditions
+                """,
+            }
+        ],
+        response_model=ModelResponse,
+    )
+
+
+if __name__ == "__main__":
+    query = """
+    Mary is an avid gardener. Yesterday, she received 18 new
+    potted plants from her favorite plant nursery. She already
+    has 2 potted plants on each of the 40 window ledges of her
+    large backyard. How many potted plants will Mary remain
+    with?
+    """
+    response = generate_response(query)
+    reconstructed_prompt = reconstruct_prompt(response)
+    print(reconstructed_prompt.reconstructed_prompt)
+    """
+    reconstructed_prompt='Mary has 40 window ledges, and each
+    ledge has 2 potted plants on it. She received 18 new
+    potted plants yesterday. How many potted plants will Mary
+    have in total?'
+    """
+
+    original_condition_list = deconstruct_prompt_into_condition_list(query)
+    new_condition_list = deconstruct_prompt_into_condition_list(
+        reconstructed_prompt.reconstructed_prompt
+    )
+    print(original_condition_list)
+    """
+    conditions = [
```

Comment:
are you adding these yourself?
the pytest-examples should be adding these but i dont know how its getting `conditions =`

Reply:
Nope, i'm just printing it out using pytest but I'll convert these to use .model_dump_json(indent=2) so that it's more consistent